### ISSUE TITLE
[Feature]: Friendly Name in Matter Hub

### ISSUE THREAD

**Issue by Hallo2024** [t0bst4r/home-assistant-matter-hub#340]
### Requested feature

After adding the entity to the Matterhub, there should be the option to assign a name that will be displayed as the device name in the Alexa app

### Documentation & Issues

- [X] I have reviewed the list of [currently supported entities and their limitations](https://github.com/t0bst4r/home-assistant-matter-hub?tab=readme-ov-file#5-supported-domains--device-types).
- [X] I have searched the [issues](https://github.com/t0bst4r/home-assistant-matter-hub/issues) for a similar requests.
- [X] I have [checked](https://github.com/project-chip/matter.js/blob/main/docs/ECOSYSTEMS.md) if my controller is known to support the requested device type.

**Comment by Senbei123** [comment]
Oh, this is a nice one. I know this feature from other Alexa-Integrations. 
As a work around you can currently rename the device in the alexa app. But to update the name directly in the matter app is more convenience.

**Comment by lukejagger** [comment]
Yes please, this will address those other issues I mentioned in the above feature request.

**Comment by wikibrikofix** [comment]
If I understand correctly the code, in the file
packages/backend/src/matter/behaviors/basic-information-server.ts
the entities are selected and for each entity is used the friendly_name and if not present, the entity_id

```
nodeLabel:
ellipse(32, entity.state?.attributes?.friendly_name) ??
ellipse(32, entity.entity_id),
```

Probably, if we apply a custom attribute to the entity

```
homeassistant:
  customize:
    light.lamp:
      matter_name: "Lamp room"
```

and we change the
packages/backend/src/matter/behaviors/basic-information-server.ts
with

```      nodeLabel:
        ellipse(32, entity.state?.attributes?.matter_name) ??
        ellipse(32, entity.state?.attributes?.friendly_name) ??
        ellipse(32, entity.entity_id),
```

and in the file
packages/common/src/home-assistant-entity-state.ts

```
export type HomeAssistantEntityStateAttributes<T extends object = object> = {
  friendly_name?: string;
  matter_name?: string;
  unit_of_measurement?: string;
  icon?: string;
```

we add the matter_name in the object
probably... we can try to have a custom name used only by the matter hub
